src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6496,6 +6503,56 @@ private BoundExpression BindLeftIdentifierOfPotentialColorColorMemberAccess(Iden
             return boundValue;
         }
 
+        private BoundExpression RebindColorColorConstField(IdentifierNameSyntax left, BoundExpression boundValue, FieldSymbol leftSymbol)
+        {
+            if (!leftSymbol.IsConst)
+            {
+                return boundValue;
+            }
+
+            var memberAccessParent = left.Parent as MemberAccessExpressionSyntax;
+            Debug.Assert(memberAccessParent is not null);
+
+            var targetBoundValue = boundValue;
+
+            var binderSymbol = ContainingType;
+            Debug.Assert(binderSymbol is TypeSymbol);
+
+            var firstOuterScopeBinder = this;
+            while (true)
+            {
+                firstOuterScopeBinder = firstOuterScopeBinder.Next;
+
+                if (firstOuterScopeBinder is null)
+                {
+                    break;
+                }
+
+                // We found the first container that does not apply to our same symbol
+                if (!binderSymbol.Equals(firstOuterScopeBinder.ContainingType))
+                {
+                    break;
+                }
+            }
+
+            // The first in container binder reflects the current scope, so we begin at the outer scope of the binder
+            for (var binder = firstOuterScopeBinder; binder is not null; binder = binder.Next)
+            {
+                // While trying to find a suitable binder for the expression, we do not report diagnostics
+                targetBoundValue = binder.BindMemberAccess(memberAccessParent, false, false, BindingDiagnosticBag.Discarded);
+
+                // We want to avoid binding to the const field symbol,
+                // so we break on the first symbol that does not equal that
+                if (!boundValue.ExpressionSymbol.Equals(targetBoundValue.ExpressionSymbol))
```

Comment:
>ExpressionSymbol [](http://example.com/codeflow?start=32&length=16)

```ExpressionSymbol``` is very "imprecise " If we assume that we are dealing with ```BoundField```s and ```FieldSymbol```s, then the code should be written in their terms.

Reply:
I didn't know you could reliably bound field expressions, perhaps I could try that out